lib/internal/util.js

```diff
@@ -186,3 +186,18 @@ exports.toLength = function toLength(argument) {
   const len = toInteger(argument);
   return len <= 0 ? 0 : Math.min(len, Number.MAX_SAFE_INTEGER);
 };
+
+// Useful for Wrapping an ES6 Class with a constructor Function that
+// does not require the new keyword. For instance:
+//   class A { constructor(x) {this.x = x;}}
+//   const B = createClassWrapper(A);
+//   B() instanceof A // true
+//   B() instanceof B // true
+exports.createClassWrapper = function createClassWrapper(type) {
+  assert(typeof type === 'function');
+  const fn = function(...args) {
+    return Reflect.construct(type, args);
```

Comment:
If you’re using `Reflect.construct` anyway, you can actually be a bit more generic and say `Reflect.construct(type, args, new.target || type)`